transformations/transformations/single_column_coalesced.py

```diff
@@ -354,15 +407,20 @@ def kernel_annotate_sequential_loops_openacc(cls, routine, horizontal):
             The subroutine in which to annotate sequential loops
         horizontal: :any:`Dimension`
             The dimension object specifying the horizontal vector dimension
+        block_dim: :any: `Dimension`
+            The dimension object specifying the blocking dimension
+        ignore: list or tuple
+            Loops to be ignored for annotation
         """
+        block_dim_index = None if block_dim is None else block_dim.index
         with pragmas_attached(routine, ir.Loop):
 
             for loop in FindNodes(ir.Loop).visit(routine.body):
                 # Skip loops explicitly marked with `!$loki/claw nodep`
                 if loop.pragma and any('nodep' in p.content.lower() for p in as_tuple(loop.pragma)):
                     continue
 
-                if loop.variable != horizontal.index:
+                if loop.variable != horizontal.index and loop.variable != block_dim_index and loop not in ignore:
```

Comment:
I agree. To be honest I thought we would determine "driver loops" via `block_dim`. However, currently we determine "driver loops" via calls to subroutines that are within `target` (and now additionally via the driver loop pragma). I don't know whether there is (or was) a reason for doing it via calls that are within `target` instead of using `block_dim`. I am open for discussion. However, switching to determining "driver loops" via `block_dim` would need some extensive testing to ensure we don't break anything ...  

Reply:
Ok, agreed, and good point! Let's leave it like this for now, and I'll file a reminder issue that we want to clean-up in a separate PR with the extensive testing.